File: backend/model/size_category.go
package model

type SizeCategory struct {
	ID        uint     `gorm:"primarykey" json:"id"`
	Name      string   `gorm:"not null" json:"name"`
	MinWeight float32  `gorm:"not null" json:"minWeight"`
	MaxWeight *float32 `json:"maxWeight"`
}

File: frontend/lib/model/size_category.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'size_category.freezed.dart';

part 'size_category.g.dart';

@freezed
class SizeCategory with _$SizeCategory {
  const factory SizeCategory({
    required int id,
    required String name,
    required double minWeight,
    required double? maxWeight,
  }) = _SizeCategory;

  factory SizeCategory.fromJson(Map<String, dynamic> json) =>
      _$SizeCategoryFromJson(json);
}
